### ISSUE TITLE
deconstructing floors being auto-suspended/unsuspended by suspendmanager

### ISSUE THREAD

**Issue by AnotherOddity** [DFHack/dfhack#3494]
A floating corridor of floors all marked for deconstruction could all be automatically suspended excluding the edge/end tile, with the next one being unsuspended after the former's deconstruction allowing safe removal without a collapse/cave-in [as mentioned here](https://www.reddit.com/r/dwarffortress/comments/14hhgm4/dfhack_5008r4_and_5009sdl22rc1_released/jpe55nk/?context=3).

**Comment by plule** [comment]
I don't believe this is possible unfortunately, deconstruction jobs cannot be suspended

![image](https://github.com/DFHack/dfhack/assets/630159/59a1d146-816d-4c60-a855-0ad0719e87ac)

**Comment by myk002** [comment]
Hrm.. what happens if you set the suspend flag anyway? In this case, it might not matter that the vanilla UI doesn't support it. We may still be able to make things safer behind the scenes

**Comment by plule** [comment]
> Hrm.. what happens if you set the suspend flag anyway? In this case, it might not matter that the vanilla UI doesn't support it. We may still be able to make things safer behind the scenes

you are right, it works ahah. no indication or way for the user to unsuspend though. Things to consider:

- Unsuspend all the non standard jobs on disabling suspendmanager
- User feedback (the existing overlay could hopefully be drawn for building removal too)

**Comment by plule** [comment]
The constraints for removing floating floors are actually remarkably similar with walls filling: deconstruct corridors one by one, but also avoid stuck corners (which would fall too)
